### PR TITLE
[compiler-v2] Relax syntax for `as` and `is` operators

### PR DESCRIPTION
**Pull request by wrwg**
## Description

Until now, one always had to put `as` and `is` in parenthesis. For example, `assert!((x is Variant))`. This PR relaxes the syntax to allow `as`/`is` on the same binding level than assignment (lowest priority among binary operators), so one can omit the parenthesis.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Added new tests

### CODE REVIEW

**Comment by wrwg** [comment]
* **#14420** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/14420?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wrwg and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: third_party/move/move-compiler-v2/tests/checking/variants/variants_test_no_parenthesis_ok.exp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    enum Generic {
+        Foo {
+            0: #0,
+        }
+        Bar {
+            0: u64,
+        }
+    }
+    private fun test(c: m::Color): bool {
+        test_variants m::Color::Red|RGB(c)
```
Comment (brmataptos) [inline third_party/move/move-compiler-v2/tests/checking/variants/variants_test_no_parenthesis_ok.exp:21]: Does `m::Color::` only apply to the left variant?  Syntactically it makes sense to be able to have a  qualifier separately on each variant, but semantically I suppose they should both be variants of the same enum.  Do we have any test cases where this qualifier is actually needed?  I don't see any.

(This applies to the old parens version, too.)

**Review by brmataptos — COMMENTED** [review]
One question about semantics/testig.

File: third_party/move/move-compiler-v2/tests/checking/variants/variants_test_no_parenthesis_ok.exp (hunk 1 above)
Comment (wrwg) [reply third_party/move/move-compiler-v2/tests/checking/variants/variants_test_no_parenthesis_ok.exp:21]: This is the debug print, and it is a bit sloppy. Should be `m::Color::{Red|RGB(c)}`. Yes indeed the compiler only allows variants of the same type, there are test cases in `variants_test_err.move` for this.

**Review by brmataptos — APPROVED** [review]

**Pull request merged**